docs/conf.py

```diff
@@ -54,6 +58,7 @@
     "sphinx_copybutton",
     # Adds a copy button to each code block in the documentation
     # https://sphinx-copybutton.readthedocs.io/en/latest/
+    "jupyter_sphinx",
```

Comment:
@nelimee What are your thoughts on making example notebooks executable? Right now, the outputs are saved and shown directly from the jupyter notebooks that were initially tested locally. 

As this is a rapidly evolving project, this will verify that the code in the examples is valid. 

One major caveat to this is that the documentation build time increases.

Reply:
I think I might have chosen the wrong example to show this as nbsphinx is used for jupyter notebooks.

But if I change the codeblocks on the `Quick Start` page, there would be a plot [here](https://tqec.github.io/tqec/user_guide/quick_start.html#plot-the-results) courtesy of the added extension

Reply:
> As this is a rapidly evolving project, this extension will verify that the code in the examples is valid. One major caveat to this is that the documentation build time increases.

Exactly what I was about to say :) I like the idea, let's see what kind of impact it has on CI times.

Reply:
Also it might be interesting to cache sinter results for the notebooks that require simulation.

Reply:
Good idea! This could help reduce the CI time.

Reply:
> Also it might be interesting to cache sinter results for the notebooks that require simulation.

@nelimee Dumb question, does sinter API have an option for this? I couldn't find it or maybe I am not looking in the right spot. 

If not, I am thinking of using [pickle](https://docs.python.org/3/library/pickle.html). There are issues with using it, though. 

![image](https://github.com/user-attachments/assets/363249eb-7b4b-4a0a-a12f-a33e1c638c9b)

Reply:
Yes, `sinter` (and `tqec.simulation.simulation.start_simulation_using_sinter`) have an argument `save_resume_filepath` that should be usable for that. 

The main "problem" here will be to cache that result in CI. It should be easy to write, but you might have permission issues. If that is the case, please let me know :)

Reply:
If I start with locally cached results, I don't think I need write permissions. See https://github.com/tqec/tqec/pull/555/commits/6a6b5f5a5af7a132c4364aa4c90e4f0856a9c755

Let me know if you had something else in mind.

Reply:
I think I ought to add some comments about why we are using `save_resume_filepath` and what a user can do if they don't want to use the cached results.

Reply:
For cache in CI, I am thinking something like that: 
https://github.com/actions/cache?tab=readme-ov-file#restoring-and-saving-cache-using-a-single-action

Reply:
> For cache in CI, I am thinking something like that: https://github.com/actions/cache?tab=readme-ov-file#restoring-and-saving-cache-using-a-single-action

Okay. I have never used a workflow like this, and it will take me a while to understand how to get it working. I plan to wrap up my other draft PR before that.

Reply:
I can have a try if you want, I already used such a workflow.

Reply:
Thanks for the offer! I want to try it first. I have never used a workflow to cache things. So, it will be a nice opportunity to learn. 

I will reach out to you with questions.

Reply:
@purva-thakre you might have issues changing workflow files, if that is the case please let me know. I will try to see what I can do to give you the appropriate rights.